bVNC/src/main/java/com/iiordanov/bVNC/Utils.java

```diff
@@ -166,6 +212,9 @@ public static void showMessage(Context _context, String title, String message, i
             builder.setIcon(icon);
             if (!(alertDialog != null && alertDialog.isShowing()) && !isContextActivityThatIsFinishing(_context)) {
                 alertDialog = builder.create();
+                if ( showHandler != null ) {
+                    alertDialog.setOnShowListener(showHandler);

```

Comment:
setOnShowListener shows its parameter is nullable, so maybe you can just pass showHandler even if it's null?